adapters/displayio/displayio.go

```diff
@@ -37,27 +37,23 @@ func (adapter *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *
 	for _, impression := range request.Imp {
 		var requestExt map[string]interface{}
 
-		if impression.BidFloor == 0 {
-			errs = append(errs, &errortypes.BadInput{
-				Message: "BidFloor should be defined",
-			})
-			continue
-		}
-
 		if impression.BidFloorCur == "" {
 			impression.BidFloorCur = "USD"
 		}
 
 		if impression.BidFloorCur != "USD" {
-			convertedValue, err := requestInfo.ConvertCurrency(impression.BidFloor, impression.BidFloorCur, "USD")
+			if impression.BidFloor != 0 {
+				convertedValue, err := requestInfo.ConvertCurrency(impression.BidFloor, impression.BidFloorCur, "USD")
 
-			if err != nil {
-				errs = append(errs, err)
-				continue
+				if err != nil {
+					errs = append(errs, err)
+					continue
+				}
+
+				impression.BidFloor = convertedValue
```

Comment:
```if impression.BidFloor != 0 && impression.BidFloorCur != "USD" {
    convertedValue, err := requestInfo.ConvertCurrency(impression.BidFloor, impression.BidFloorCur, "USD")
    if err != nil {
        errs = append(errs, err)
        continue
    }
    impression.BidFloor = convertedValue
    impression.BidFloorCur = "USD"
}```
Combining the two checks (BidFloor != 0 and BidFloorCur != "USD") into a single if block makes the code more concise and avoids excessive nesting. This improves readability and understanding of the logic because: It expresses in one condition that currency conversion only occurs when the currency is not USD and BidFloor is not zero.

Reply:
Hi @xdevel, thanks for pointing that out – I understand now that BidFloorCur needs to be set to "USD" regardless of whether BidFloor is zero or not.

Based on this requirement, here’s the updated code I propose to ensure BidFloorCur is set to "USD" and currency conversion only occurs if BidFloor is greater than zero:
```if impression.BidFloor != 0 && impression.BidFloorCur != "USD" {
    convertedValue, err := requestInfo.ConvertCurrency(impression.BidFloor, impression.BidFloorCur, "USD")
    if err != nil {
        errs = append(errs, err)
        continue
    }
    impression.BidFloor = convertedValue
    impression.BidFloorCur = "USD"
} else if impression.BidFloorCur != "USD" {
    impression.BidFloorCur = "USD"
}
```